AdyenEncryption/JOSE/Encryption Algorithms/JSON Web Encryption/JSONWebEncryptionGenerator.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+internal protocol AnyJSONWebEncryptionGenerator {
+    func generate(withPayload: Data,
+                  publicRSAKey: SecKey,
+                  mockInitializationVector: Data?,
+                  mockContentEncryptionKey: Data?,
+                  header: JSONWebEncryption.Header) throws -> JSONWebEncryption
+}
+
+internal struct JSONWebEncryptionGenerator: AnyJSONWebEncryptionGenerator {
+    internal func generate(withPayload: Data,
+                           publicRSAKey: SecKey,
+                           mockInitializationVector: Data? = nil,
+                           mockContentEncryptionKey: Data? = nil,
+                           header: JSONWebEncryption.Header) throws -> JSONWebEncryption {
+        let keyEncryptionAlgorithm = header.keyEncryptionAlgorithm.algorithm
```

Comment:
It is probably a good idea to provide `algorithm` as a parameter - for future scalability?